acceptance/ui/features/steps/user_steps.rb

```diff
@@ -0,0 +1,39 @@
+When(/^I view user details$/) do
+    viewUserDetails()
+end
+
+When (/^I clear my messages$/) do
+  page.find("[ng-click='clearMessages()']").click()
+end
+
+When (/^I click on an unread message$/) do    # clicks on the first unread message
+  Capybara.match=:first
+  page.find("[class='message unreadMessage ng-scope']").click()
+  Capybara.match=:smart
+end
+
+When (/^I switch the language to English$/) do
+  page.find("input[value='en_US']").click()
+end
+
+When (/^I switch the language to Spanish$/) do
+  page.find("input[value='es_US']").click()
+end
+
+Then /^I should see my messages$/ do
+  page.assert_selector("[ng-repeat='message in messages.messages track by message.id']")
+end
+
+Then /^I should not see any messages$/ do
+  page.assert_no_selector("[ng-repeat='message in messages.messages track by message.id']")
+end
+
+Then /^I should see a checkmark$/ do
```

Comment:
This step definition reads like something really generic, but the implementation is very specific to this page.
How about something like `Then I should see that the message is marked as read`